src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -180,7 +180,8 @@ public override Binder VisitMethodDeclaration(MethodDeclarationSyntax methodDecl
                         resultBinder = new InMethodBinder(method, resultBinder);
                     }
 
-                    resultBinder = resultBinder.WithUnsafeRegionIfNecessary(methodDecl.Modifiers);
+                    resultBinder = resultBinder.WithUnsafeRegionIfNecessary(methodDecl.Modifiers,
+                        isIteratorBody: usage == NodeUsage.MethodBody && method?.IsIterator == true);
```

Comment:
>? [](http://example.com/codeflow?start=79&length=1)

It looks like the conditional access is unnecessary. Also, consider introducing a boolean local and setting it accordingly inside ```if (usage == NodeUsage.MethodBody)``` above instead of calculating the value here trying to carefully duplicate conditions under which we are creating an in method binder.